ApplicationLibCode/ProjectDataModel/Summary/RimMultipleSummaryPlotNameHelper.cpp

```diff
@@ -0,0 +1,257 @@
+//
+//  ResInsight is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  ResInsight is distributed in the hope that it will be useful, but WITHOUT ANY
+//  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//  FITNESS FOR A PARTICULAR PURPOSE.
+//
+//  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+//  for more details.
+//
+/////////////////////////////////////////////////////////////////////////////////
+
+#include "RimMultipleSummaryPlotNameHelper.h"
+
+#include <QString>
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimMultiSummaryPlotNameHelper::RimMultiSummaryPlotNameHelper( std::vector<const RimSummaryPlotNameHelperInterface*> nameHelpers )
+    : m_nameHelpers( nameHelpers )
+{
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+QString RimMultiSummaryPlotNameHelper::plotTitle() const
+{
+    if ( m_nameHelpers.size() == 1 ) return m_nameHelpers.front()->plotTitle();
+
+    if ( m_nameHelpers.size() == 2 )
+    {
+        auto first  = m_nameHelpers[0];
+        auto second = m_nameHelpers[1];
+
+        return first->aggregatedPlotTitle( *second );
+    }
+
+    return "Plot Title";
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RimMultiSummaryPlotNameHelper::isPlotDisplayingSingleQuantity() const
+{
+    int plotCountWithSingleQuantity = 0;
+    for ( auto nameHelper : m_nameHelpers )
+    {
+        if ( nameHelper->isPlotDisplayingSingleQuantity() ) plotCountWithSingleQuantity++;
+    }
+
+    if ( plotCountWithSingleQuantity == 1 ) return true;
+
+    return false;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RimMultiSummaryPlotNameHelper::isWellNameInTitle() const
+{
+    for ( auto nameHelper : m_nameHelpers )
+    {
+        if ( nameHelper->isWellNameInTitle() ) return true;
+    }
+    return false;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RimMultiSummaryPlotNameHelper::isWellGroupNameInTitle() const
+{
+    for ( auto nameHelper : m_nameHelpers )
```

Comment:
`return std::any_of(foo.begin(), foo.end(), [](auto nameHelper){return nameHelper->isWellGroupNameInTitle()}`

More "functional" style. Not sure if it is better.. :-)